src/dicomweb_client/api.py

```diff
@@ -1357,16 +1411,53 @@ def _http_post(
 
         '''
         logger.debug('POST: {} {}'.format(url, headers))
-        response = self._session.post(url=url, data=data, headers=headers)
+
+        def serve_data_chunks(data):
+            for i, offset in enumerate(range(0, len(data), self._chunk_size)):
+                end = offset + self._chunk_size
+                yield data[offset:end]
+
+        if self._chunk_size is not None and len(data) > self._chunk_size:
+            logger.info('store data in chunks using chunked transfer encoding')
+            chunked_headers = dict(headers)
+            chunked_headers['Transfer-Encoding'] = 'chunked'
+            chunked_headers['Cache-Control'] = 'no-cache'
+            chunked_headers['Connection'] = 'Keep-Alive'
```

Comment:
`chunked_headers` isn't being used, the subsequent request uses the original `headers`. Also, according to `requests`' documentation on [chunk-encoded requests](https://2.python-requests.org/en/v1.2.3/user/advanced/#chunk-encoded-requests), passing an iterator is all that's required so it seems like `Trasfer-Encoding: chunked` is taken care of automatically by `requests`.

Reply:
Thanks for catching this. I created an issue: #25